internal/providers/terraform/azure/application_gateway.go

```diff
@@ -20,13 +20,16 @@ func GetAzureRMApplicationGatewayRegistryItem() *schema.RegistryItem {
 
 func NewAzureRMApplicationGateway(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
 	region := lookupRegion(d, []string{})
-	var monthlyDataProcessedGb, monthlyCapacityUnits *decimal.Decimal
+	var monthlyDataProcessedGb *decimal.Decimal
 	skuName := d.Get("sku.0.name").String()
 	var sku, tier string
 	costComponents := make([]*schema.CostComponent, 0)
 	tierLimits := []int{10240, 30720}
 
-	capacity := d.Get("sku.0.capacity").Int()
+	capacityUnits := d.Get("sku.0.capacity").Int()
```

Comment:
Should this take precedence over usage value when it's > 0? It's optional for autoscaling, so that's where the usage value can be handy. However, if this is set explicitly (for example = 2), can it go over to the usage value?

Also, when the capacity attribute is missing and the usage value is not set, should we default `capacityUnits` to 1? For this case, we need another example with `autoscale_configuration` [block](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/application_gateway#autoscale_configuration).

Reply:
Yep, updated to be that precedence, and also use the `autoscale_configuration > min_capacity` if nothing else is provided.